src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEquals.cs

```diff
@@ -13,17 +13,48 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
+    /// <summary>
+    /// A strongly-typed `public bool Equals(T other)` method.
+    /// There are two types of strongly-typed Equals methods:
+    /// the strongly-typed virtual method where T is the containing type; and
+    /// overrides of the strongly-typed virtual methods from base record types.
+    /// </summary>
     internal sealed class SynthesizedRecordEquals : SynthesizedInstanceMethodSymbol
     {
+        private readonly PropertySymbol _equalityContract;
+        private readonly MethodSymbol? _otherEqualsMethod;
+        private readonly int _memberOffset;
+
         public override NamedTypeSymbol ContainingType { get; }
 
-        public SynthesizedRecordEquals(NamedTypeSymbol containingType)
+        public SynthesizedRecordEquals(
+            NamedTypeSymbol containingType,
+            TypeSymbol parameterType,
+            bool isOverride,
+            PropertySymbol equalityContract,
+            MethodSymbol? otherEqualsMethod,
+            int memberOffset)
         {
+            var compilation = containingType.DeclaringCompilation;
+            bool isStruct = parameterType.IsStructType();
```

Comment:
Given that records are emitted as `class` when would this be `true`?

Reply:
It cannot be `true` currently. I left the code here since the code existed before this PR but I can remove it.

Reply:
Looks like we're green now with two sign offs. I'd say go for the merge now and remove this later. Easy to squeeze in as a bug fix later.

Reply:
Removed in https://github.com/dotnet/roslyn/pull/44912.